Cargo.toml

```diff
@@ -24,4 +24,4 @@ rust-version = "1.64"
 [workspace.dependencies]
 diff = { version = "0.1.12", default_features = false }
 regex = { version = "1.8", default_features = false, features = ["std"] }
-regex-syntax = { version = "0.6", default_features = false, features = ["unicode"] }
```

Comment:
Until we get https://github.com/lalrpop/lalrpop/issues/789, I think we have to keep specifying unicode on regex-syntax.

In particular, enabling the feature for our local testing, but not for users is a recipe to mask issues around feature underspecification.

The problem here is that users can specify a regex with unicode characters without directly depending on regex-syntax.  Lalrpop then hands those characters to regex-syntax, resulting in errors.  Users should not need to know that lalrpop depends on regex-syntax, and add a dependency on it to enable its features.

The real answer is #789, but in the interim, I think it's broken to pass on valid user input to regex-syntax and then expect users to look into lalrpop internals and set a feature on a dependency they wouldn't otherwise include at all.

In particular this causes problems related to versioning.  Lalrpop users needing unicode support are currently working around by setting the unicode feature on regex-syntax 0.6.  When lalrpop updates to regex-syntax 0.7, those users 1. Will have an unnecessary 0.6 dependency and 2. Will be once again underspecified on regex-syntax 0.7.

Reply:
> In particular, enabling the feature for our local testing, but not for users is a recipe to mask issues around feature underspecification.

Yes. Tests should not require optional features (unless they are testing things related to that feature).
The issue with testing is that they nearly all require such unicode feature. 

I personally don't like having to enable features for a transitive dependency.

I can introduce this feature here, that's not really complicated.
Some of the tests need to be disabled when unicode is not set because they make use of `(?i)` an this eventually gets passed down to `regex` (not all tests using `(?i)` require the regex features).

PS: As I was doing this, I identified another issue (compounding with the previous) that made features even harder to identify.
The workspace does not set `resolver = "2"` so the features used in `build-dependencies` were leaking in the `dependencies`.
Expect a new commit soon™ covering all this.

Reply:
Is this resolved by adding `unicode` feature in this PR?

Reply:
I think so. `unicode` on `regex-syntax` gets enabled as required by enabling `unicode` on `lalrpop`.